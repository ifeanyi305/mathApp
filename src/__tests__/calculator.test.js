import { render } from "@testing-library/react";
import CalcInterface from '../components/calculator';

describe("Test component", () => {
    it("Renders correctly", () => {
      const tree = render(<CalcInterface />);
      expect(tree).toMatchSnapshot();
    });
});
