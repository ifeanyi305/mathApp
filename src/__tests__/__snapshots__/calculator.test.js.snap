// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="practice"
        >
          <p>
            let's do some maths
          </p>
        </div>
        <div
          class="calc-container"
        >
          <div
            class="header"
          >
            <button
              class="p-tag"
              type="button"
            >
              0
            </button>
          </div>
          <div
            class="calc-buttons"
          >
            <button
              type="submit"
            >
              AC
            </button>
            <button
              type="submit"
            >
              +/-
            </button>
            <button
              type="submit"
            >
              %
            </button>
            <button
              class="color"
              type="submit"
            >
              รท
            </button>
            <button
              type="submit"
            >
              7
            </button>
            <button
              type="submit"
            >
              8
            </button>
            <button
              type="submit"
            >
              9
            </button>
            <button
              class="color"
              type="submit"
            >
              x
            </button>
            <button
              type="submit"
            >
              4
            </button>
            <button
              type="submit"
            >
              5
            </button>
            <button
              type="submit"
            >
              6
            </button>
            <button
              class="color"
              type="submit"
            >
              -
            </button>
            <button
              type="submit"
            >
              1
            </button>
            <button
              type="submit"
            >
              2
            </button>
            <button
              type="submit"
            >
              3
            </button>
            <button
              class="color"
              type="submit"
            >
              +
            </button>
            <button
              class="span"
              type="submit"
            >
              0
            </button>
            <button
              type="submit"
            >
              .
            </button>
            <button
              class="color"
              type="submit"
            >
              =
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="practice"
      >
        <p>
          let's do some maths
        </p>
      </div>
      <div
        class="calc-container"
      >
        <div
          class="header"
        >
          <button
            class="p-tag"
            type="button"
          >
            0
          </button>
        </div>
        <div
          class="calc-buttons"
        >
          <button
            type="submit"
          >
            AC
          </button>
          <button
            type="submit"
          >
            +/-
          </button>
          <button
            type="submit"
          >
            %
          </button>
          <button
            class="color"
            type="submit"
          >
            รท
          </button>
          <button
            type="submit"
          >
            7
          </button>
          <button
            type="submit"
          >
            8
          </button>
          <button
            type="submit"
          >
            9
          </button>
          <button
            class="color"
            type="submit"
          >
            x
          </button>
          <button
            type="submit"
          >
            4
          </button>
          <button
            type="submit"
          >
            5
          </button>
          <button
            type="submit"
          >
            6
          </button>
          <button
            class="color"
            type="submit"
          >
            -
          </button>
          <button
            type="submit"
          >
            1
          </button>
          <button
            type="submit"
          >
            2
          </button>
          <button
            type="submit"
          >
            3
          </button>
          <button
            class="color"
            type="submit"
          >
            +
          </button>
          <button
            class="span"
            type="submit"
          >
            0
          </button>
          <button
            type="submit"
          >
            .
          </button>
          <button
            class="color"
            type="submit"
          >
            =
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
